tests/robot-tests/args_and_variables.py

```diff
@@ -126,12 +126,22 @@ def create_argument_parser() -> argparse.ArgumentParser:
         default=None,
         help="manually specify the expiredinvite user password",
     )
+    parser.add_argument(
+        "--reseed",
+        dest="reseed",
+        action="store_true",
+        help="run the seed data generation scripts against the target environment",
+    )
 
     return parser
 
 
 def initialise() -> argparse.Namespace:
     args = create_argument_parser().parse_args()
+
+    if args.reseed and args.env == "prod":
+        raise Exception(f"Cannot generate seed data against environment {args.env}")
```

Comment:
Yup sounds like a good idea.  I think I didn't do this before because of both this test and run_tests.py's dependency on `data_changing_tests`.

I have now moved the logic of calculating `data_changing_tests` into `args_and_variables.py` itself, and referenced it for both callsites now, ta.